### DISCUSSION TITLE
Can I get the data for a notification received while my Android App is in the Background?

### DISCUSSION THREAD

**Discussion by danielzen** [capawesome-team/capacitor-firebase#503] [Q&A] [answered]
From all of my reading, in order to get this information on the Android side, from the `onCreate()` of the Activity I would need to:

```
getActivity().getIntent().getExtras()
```

Unfortunately, I am not sure how to implement that in my Capacitor app, and I am wondering if it is something that can be incorporated into the `messaging` plugin?

At the moment, I am unable to get the title, body, or image from an Android notification when it arrives in the background, and not for a lack of trying.

Is there anything that can be implemented to help make this easier for Capacitor/Ionic? 🙂

**Comment by robingenz** [comment, marked as answer]
According to the documentation of the [addListener('notificationReceived', ...)](https://capawesome.io/plugins/firebase/cloud-messaging/#addlistenernotificationreceived) method:

> On Android, this listener is called for every push notification if the app is in the foreground. If the app is in the background, then this listener is only called on data push notifications. See https://firebase.google.com/docs/cloud-messaging/android/receive#handling_messages for more information.

So yes, you can receive notifications in the background. Make sure that you send a data push notification.

**Reply by danielzen** [reply]
Thanks, that does work. I guess it's just annoying that I can't do that with targeted test messages from the FCM Compose Notification window. And, that this isn't an issue with iOS, or even Web Notifications for that matter.

Thanks for all the hard work on the plugin. Much appreciated.
